pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisStreamMetadataProvider.java

```diff
@@ -108,8 +108,6 @@ public StreamPartitionMsgOffset fetchStreamPartitionOffset(OffsetCriteria offset
       if (shard == null) { // Shard has expired
         shardsEnded.add(shardId);
         continue;
-        // FIXME: Here we assume that we were done consuming the shard before it expired.
-        //  Handle edge case where consumer lags behind, resulting in shard to expire before it is all consumed
```

Comment:
I would prefer to have a log statement or emit a metric here for serviceability

Reply:
Ok. What would you prefer, A LOG to just mark an expiring shard or a LOG to warn users that although we are expiring a shard the data is not consumed completely from it and hence might be some data loss?

Makes sense to LOG the second action but it will reflect poorly on the database since there is no inbuilt solution to this problem.

Reply:
The only thing worse than data loss is concealing data loss when it happens. If there is a risk of data loss it should be logged when it happens and documented clearly.

Reply:
Added the log. Implementing an exact check if data loss has occurred or not won't be possible in this class since there is no information about what is the last sequence id (i.e. offset) in the expired shard.